src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java

```diff
@@ -198,15 +199,17 @@ public BlockPos getPlayerToFollow()
         }
         task = Task.GUARD;
         return this.getLocation();
-    }    @Override
+    }
+
+    @Override
     public void onUpgradeComplete(final int newLevel)
     {
-        for (final EntityCitizen citizen : getWorkerEntities())
+        if (newLevel > MAX_VISION_BONUS_MULTIPLIER)
         {
-            if (newLevel > MAX_VISION_BONUS_MULTIPLIER)
-            {
-                citizen.getEntityAttribute(SharedMonsterAttributes.MAX_HEALTH).setBaseValue(SharedMonsterAttributes.MAX_HEALTH.getDefaultValue() + getBonusHealth());
-            }
+            getWorkerEntities().stream()
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+              .forEach(entityCitizen -> entityCitizen.getEntityAttribute(SharedMonsterAttributes.MAX_HEALTH).setBaseValue(SharedMonsterAttributes.MAX_HEALTH.getDefaultValue() + getBonusHealth()));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this 196 characters long line (which is greater than 180 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)